controllers/database_controller.go

```diff
@@ -108,7 +111,9 @@ func (r *DatabaseReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	// To check and handle the case when the database ha been deleted/aborted externally (not through the operator).
 	err = r.handleExternalDelete(ctx, database, ndbClient)
 	if err != nil {
-		log.Error(err, "Error occurred while external delete check")
+		errStatement := "Error occurred while external delete check"
+		log.Error(err, errStatement)
+		r.recorder.Eventf(database, "Warning", "InternalError", errStatement+"Error: %s.", err.Error())
```

Comment:
Two points to consider:
1). do we want to say 'InternalError' ? 
I know we return this in NDB's APIs but do we want to return it in the operator as well? Or should we use a more meaningful term (like we have  `DatabaseCRDeletionStarted` below).

2). The 2nd argument says "Warning" and the 3rd says "InternalError". Are we returning an error or a warning to the customer?

Reply:
Removed this event, as I am recording events inside the handleDelete function. So this ambiguous event wont be there.